clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -4276,13 +4276,18 @@ EmitConditionalOperatorLValue(const AbstractConditionalOperator *expr) {
     return EmitUnsupportedLValue(expr, "conditional operator");
   rhsBlock = Builder.GetInsertBlock();
 
-  EmitBlock(contBlock);
-
   if (lhs && rhs) {
-    llvm::PHINode *phi = Builder.CreatePHI(lhs->getPointer()->getType(),
-                                           2, "cond-lvalue");
-    phi->addIncoming(lhs->getPointer(), lhsBlock);
-    phi->addIncoming(rhs->getPointer(), rhsBlock);
+    llvm::Value *lhsPtr = lhs->getPointer();
+    llvm::Value *rhsPtr = rhs->getPointer();
+    if (rhsPtr->getType() != lhsPtr->getType())
+      rhsPtr = Builder.CreatePointerBitCastOrAddrSpaceCast(rhsPtr,
```

Comment:
This is not always legal. If RHS is a local pointer and LHS is a global pointer, the cast is "local->global", which is invalid.
If pointer address spaces are different, the only correct cast is a cast to generic (addrspace 4). Check the #388, where a similar change was made around `select` instruction.

Reply:
Added casts to generic address space as needed.